GLOSSARY.md

```diff
@@ -0,0 +1,44 @@
+| MVCC        | Multi-Version Concurrency Control                                      |
+| PMR         | Polymorphic Memory Resource (see types.hpp)                            |
+| PosList     | Position List (as used in ReferenceColumns)                            |
+| ref_column  | Reference Column                                                       |
+| tid         | Transaction ID (used for MVCC)                                         |
+| val_column  | Value Column                                                           |
+
+# Glossary
+
+This explains high-level terms that have a specific meaning within Hyrise and that are used all over the code base. We do not want to document every class name here.
+
+| Name                       | Description                                                                                                                                                                |
+| -------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Abstract Syntax Tree (AST) | The old name of the *Logical Query Plan*. It was renamed when the LQP stopped being a tree and became a DAG. Shoot this old term on sight.                                 |
+| Expression                 | Any type of SQL expression that is either logical or creates a new column (e.g., `col_a = 3`, `col_a + 4`).                                                                |
+| Iterable                   | A good way to iterate over different types of columns. See `base_iterators.hpp` to start.                                                                                  |
+| Logical Query Plan         | The logical representation of a query plan. It includes nodes like "Predicate" (a filter) or "Join" but does not describe what implementation is used for executing these. |
+| Node (LQP)                 | A node in a logical query plan.                                                                                                                                            |
+| Node (NUMA)                | A NUMA memory node (usually a CPU socket).                                                                                                                                 |
+| Physical Query Plan        | The physical representation of a query plan. It holds the actual *operators* used to execute the query. Usually created by a translator from the LQP.                      |
+| Operator                   | A class that usually takes 0-2 input tables and creates one output table. Example: TableScan, Join, Input.                                                                 |
+| Rule                       | A rule in our optimizer. For instance, pushing all predicates as far down as possible is a rule.                                                                           |
```

Comment:
Should we call this Optimizer Rule?

Reply:
My idea was that whenever you see a term in the wild, you can look it up here. The files / class names only have rule in it, not optimizer rule.

Reply:
Nice